### DISCUSSION TITLE
Discussion: Use XAML Styler for consistency?

### DISCUSSION THREAD

**Discussion by michael-hawker** [microsoft/microsoft-ui-xaml#7455] [General]
# Discussion: Use XAML Styler for consistency?
[XAML Styler](https://github.com/Xavalon/XamlStyler) is a tool to help make XAML consistent by applying a common set of style rules to XAML (i.e. VSM should be on top, or Grid Row/Columns should come first, etc...)

It'd be great if the repo could adopt a set of guidance for XAML styling, encoded in a XAML Styler settings file. This would ensure:
- XAML templates are consistent across the repo
- XAML templates are easier to read (as they are consistently organized)
- Other projects could adopt the same settings for consistency with platform styles

The last point would be especially helpful for libraries and app developers which copy and re-template controls, as if there's a future update to a style, since the ordering of everything could be enforced to be consistent (between WinUI and an app developer adopting the same style settings), then this would make it easier to compare the styles for differences to update between versions.

## Related Links
- #3735 
- #2464
- #3939
- #279

**Comment by StephenLPeters** [comment]
Love this, @karenbtlai was also interested.

**Comment by beervoley** [comment]
that would be awesome :)

**Comment by StephenLPeters** [comment]
@ranjeshj FYI
